app/controllers/admin_shelters_controller.rb

```diff
@@ -2,4 +2,8 @@ class AdminSheltersController < ApplicationController
   def index
     @shelters = Shelter.find_by_sql("SELECT * FROM shelters ORDER BY shelters.name DESC")
   end
-end
\ No newline at end of file
+
+  def show
+    @shelters = Shelter.find_by_sql("SELECT * FROM shelters ORDER BY shelters.name DESC")
```

Comment:
Since it's a show page, it should only have data for one shelter (the one specified in the params). The story also only wants you to get the name and address. So you need to add an address to the shelters table, then use SQL like this: `@shelter = Shelter.find_by_sql("SELECT name, address FROM shelters WHERE id = ?", params[:id])`